crates/driver/src/settlement_proposal.rs

```diff
@@ -0,0 +1,356 @@
+use crate::commit_reveal::SettlementSummary;
+use anyhow::{Context, Result};
+use model::order::{Order, OrderKind};
+use num::{BigRational, ToPrimitive};
+use primitive_types::{H160, U256};
+use shared::{
+    conversions::U256Ext, http_solver::model::TokenAmount, price_estimation::gas::GAS_PER_ORDER,
+};
+use solver::settlement::{
+    external_prices::ExternalPrices, trade_surplus_in_native_token, verify_executed_amount,
+    Interaction, Settlement, SettlementEncoder, TradeExecution,
+};
+use std::{
+    collections::hash_map::{Entry, HashMap},
+    sync::Arc,
+};
+
+/// Data describing the side effects of an on-chain interaction without the finalized call data.
+#[derive(Debug, Clone)]
+pub struct InteractionMetadata {
+    inputs: Vec<TokenAmount>,
+    outputs: Vec<TokenAmount>,
+    gas_used: U256,
```

Comment:
Why do we include gas used? (Not saying it isn't needed but I can't think of why it is necessary between the `autopilot` <-> `driver`).

Reply:
Ah 🤦 - this is used internally for RFQ interactions that need indicative gas consumption.